__tests__/server/controllers/dataController.test.ts

```diff
@@ -0,0 +1,113 @@
+import { Request, Response, NextFunction } from 'express';
+import { determineClusters, Cluster, TreeNode } from '../../../src/server/utils/endpoints';
+import { getClusterList, getClusterLoadGraphData, getClusterTreeGraphData } from './../../../src/server/controllers/dataController';
+import { connect, disconnect } from '../../../src/shared/models/mongoSetup';
+import { EndpointModel, Endpoint} from './../../../src/shared/models/endpointModel';
+import { simulateServerResponses, EndpointPDF, DistributionFunction } from './../../../src/shared/utils/dataGenerator';
+import { LoadData } from '../../../src/server/utils/endpoints';
+
+describe('Testing middleware', () => {
+  let mockRequest: Partial<Request>;
+  let mockResponse: Partial<Response>;
+  let mockNext: NextFunction;
+
+  beforeAll(async () => {
+    console.log('About to connect');
+    await connect('mongodb+srv://admin:test@cluster0.dopf4.mongodb.net/DePaul?retryWrites=true&w=majority');
+    console.log('Connection should have been made');
+  });
+
+  afterAll(async () => {
+    await EndpointModel.deleteMany();
+    await disconnect();
+  });
+
+  beforeEach(async () => {
+    mockRequest = {};
+    mockResponse = {
+      locals: {},
+    };
+    mockNext = jest.fn();
+    await EndpointModel.deleteMany();
+  });
+
+  test('getClusterList: Retrieves correct method and endpoints', async () => {
+
+    const endpointsPDF: EndpointPDF[] = [
+      { method: 'GET', endpoint: '/api/1', pdf: () => 1 / 24 },
+      { method: 'GET', endpoint: '/api/2', pdf: (x) => x / 24 },
+      { method: 'GET', endpoint: '/api/3', pdf: (x) => x / 24 },
+      { method: 'POST', endpoint: '/api/4', pdf: () => 1 / 24 }
+    ];
+    const callDist: DistributionFunction = () => 100;
+    const singleEndpointServerResponses: Endpoint[] = simulateServerResponses(endpointsPDF, callDist, 5, 60);
```

Comment:
You call this variable singleEndpointServerResponse, but 4 different endpoints are passed in.